aptos-move/aptos-release-builder/src/components/framework.rs

```diff
@@ -85,27 +89,46 @@ pub fn generate_upgrade_proposals(
         // 1-aptos-stdlib.move	3-aptos-token.move	5-version.move		7-consensus-config.move
         // The first framework file being generated is 3-aptos-token.move. It's using the next_execution_hash being passed in (so in this case, the hash of 4-gas-schedule.move being passed in mod.rs).
         // The second framework file being generated would be 2-aptos-framework.move, and it's using the hash of 3-aptos-token.move (which would be result.last()).
-        let mut _next_execution_hash_string = "".to_owned();
-        if !next_execution_hash.clone().is_empty() {
-            args.push("--next-execution-hash");
-            // Convert from bytes to string to pass next_execution_hash to the command line
-            if result.is_empty() {
-                _next_execution_hash_string = hex::encode(next_execution_hash.clone());
-            } else {
-                _next_execution_hash_string = hex::encode(get_execution_hash(&result));
-            }
-            args.push(&_next_execution_hash_string);
-        }
-
-        assert!(Command::new("cargo")
-            .current_dir(root_path.as_path())
-            .args(args)
-            .output()
-            .unwrap()
-            .status
-            .success());
 
-        let script = std::fs::read_to_string(move_script_path.as_path())?;
+        let options = BuildOptions {
+            with_srcs: true,
+            with_abis: false,
+            with_source_maps: false,
+            with_error_map: true,
+            skip_fetch_latest_git_deps: false,
+            bytecode_version: Some(config.bytecode_version),
+            ..BuildOptions::default()
+        };
+        let package = BuiltPackage::build(package_path, options)?;
+        let release = ReleasePackage::new(package)?;
+
+        // If we're generating a single-step proposal on testnet
+        if is_testnet && next_execution_hash.is_empty() {
```

Comment:
I wonder if we should just get rid of this case once we have support for resolution through the standard voting process in testnet. This would also allow us to deprecated and delete the code that generate the proposals with the get_signer_testnet function as we'd no longer need it

Reply:
Yes I'm working on that rn.